merchant/src/main.rs

```diff
@@ -166,7 +166,10 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
         .unwrap_or(DEFAULT_DB_FILE.into());
     let port = std::env::var("PORT").unwrap_or(DEFAULT_PORT.into());
 
-    let conn = Pool::connect(&db_file).await?;
+    let conn = PoolOptions::new()
+        .max_connections(1)
```

Comment:
Had to set this because sqlx is not flushing between database calls. See https://github.com/launchbadge/sqlx/issues/1648#issuecomment-1019387449. To fix it completely we need to share the same connection for all db operations during the request lifetime.

Reply:
this is workaround for this moment and we can proceed with the merge, what do you think @olanod ?